src/main/java/algorithms/curated170/medium/ConvertBinarySearchTreeToDoublyLinkedList.java

```diff
@@ -0,0 +1,44 @@
+package algorithms.curated170.medium;
+
+import algorithms.datastructures.TreeNode;
+
+public class ConvertBinarySearchTreeToDoublyLinkedList {
+    private TreeNode headPointer = new TreeNode(0);
+    private TreeNode lastNode = headPointer;
+
+    public TreeNode treeToDoublyList(TreeNode root) {
+        convertTree(root);
+        lastNode.right = headPointer.right;
+        headPointer.right.left = lastNode;
```

Comment:
We need to add a null-check before using **headPointer.right** as there is a possible case in which **headPointer.right** has never been populated.

Reply:
Quite a nice solution using a "sentinel node" as a dummy head.
I approve already.
To make it clear for the future audience, I draw your code as much as I understand it:

![IMG-1594](https://user-images.githubusercontent.com/60903744/117470195-3623c580-af5f-11eb-8b02-d06510ed3e85.jpg)

Reply:
Does the new null check fix this problem?

Reply:
Yep, it fixes like this:
(LeetCode uses Node instead of TreeNode, by the way. 
 You can also import Node using "import algorithms.datastructures.Node"
 
![image](https://user-images.githubusercontent.com/60903744/117477276-7aff2a80-af66-11eb-8230-40ce7bbdc643.png)